### ISSUE TITLE
Add Code Unit Test Coverage To Project

### ISSUE THREAD

**Issue by kb1lqc** [FaradayRF/faradayio#30]
Try implemented `coveralls` to show the code coverage of our unit tests.

https://coveralls.io/

This should also integrate with travis-ci as much as possible.

https://docs.travis-ci.com/user/coveralls/

I'm guessing we should use the largest of the two python options:
https://github.com/coveralls-clients/coveralls-python

This one has the latest commit and larger stars on github. Arbitrary but I like using something better supported/used.

Tasks:

- [x] Added coverage support with `coveralls` and `coverage`
- [x] Add coverage badge to `readme.md`

**Comment by kb1lqc** [comment]
OK I have something working I guess. I used http://www.robinandeer.com/blog/2016/06/22/how-i-test-my-code-part-3/ to help me run pytest-cov and called the `faradayio` package to get the following report.

```
sudo .venv/bin/pytest --cov-report html --cov faradayio
```

![image](https://user-images.githubusercontent.com/331540/35478617-e7502754-0396-11e8-9bf8-882ee44b87fd.png)

I am a bit confused that I run `faradayio` without reall having a package built. It hopefully is catching the entire module.

**Comment by kb1lqc** [comment]
Removing the HTML output I get a nice command line feedback:
![image](https://user-images.githubusercontent.com/331540/35478628-3943ddee-0397-11e8-849f-30f4a362d0ca.png)

**Comment by kb1lqc** [comment]
OK I should move this to travis-ci

**Comment by kb1lqc** [comment]
Also got this to work with `coveralls`:

```
sudo .venv/bin/python3 -m coverage run .venv/bin/pytest
```

![image](https://user-images.githubusercontent.com/331540/35478675-75118f5a-0398-11e8-8ead-d0b2f190f8b2.png)

**Comment by kb1lqc** [comment]
It worked! Looks like it is as simple as adding the coveralls test line. It does appear I am counting the root directory as well which is really bringing down the code coverage value. This seems like I'm missing a configuration value.

https://travis-ci.org/kb1lqc/faradayio/builds/334512896

![image](https://user-images.githubusercontent.com/331540/35488968-5d281b22-0445-11e8-8c7f-46f2b44ce20e.png)


Here's a coverage badge on this build:
[![Coverage Status](https://coveralls.io/repos/github/kb1lqc/faradayio/badge.svg?branch=coveralls)](https://coveralls.io/github/kb1lqc/faradayio?branch=coveralls)
![image](https://user-images.githubusercontent.com/331540/35488974-7409069e-0445-11e8-8816-9233d2e514cd.png)

**Comment by kb1lqc** [comment]
Alright specifying the source directory for `faradayio` in `.coveragerc` worked!
https://coveralls.io/builds/15260517

![image](https://user-images.githubusercontent.com/331540/35489106-4a5f3726-0447-11e8-8f4b-3b774a780b5b.png)

**Comment by kb1lqc** [comment]
Merged into `develop` per kb1lqc/faradayio#3
